server/channels/store/sqlstore/autotranslation_store.go

```diff
@@ -0,0 +1,443 @@
+		Values(objectType, objectID, dstLang, providerID, normHash, text, confidence, json.RawMessage(metaBytes), now, contentSearchText).
+		Suffix(`ON CONFLICT (object_type, object_id, dst_lang)
+				DO UPDATE SET
+					provider_id = EXCLUDED.provider_id,
+					norm_hash = EXCLUDED.norm_hash,
+					text = EXCLUDED.text,
+					confidence = EXCLUDED.confidence,
+					meta = EXCLUDED.meta,
+					updateat = EXCLUDED.updateat,
+					content_search_text = EXCLUDED.content_search_text
+					WHERE translations.norm_hash IS DISTINCT FROM EXCLUDED.norm_hash`)
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return model.NewAppError("SqlAutoTranslationStore.Save",
+			"store.sql_autotranslation.query_build_error", nil, err.Error(), 500)
+	}
+
+	if _, err := s.GetMaster().Exec(queryString, args...); err != nil {
+		return model.NewAppError("SqlAutoTranslationStore.Save",
+			"store.sql_autotranslation.save.app_error", nil, err.Error(), 500)
+	}
+
+	return nil
+}
+
+func (s *SqlAutoTranslationStore) Search(dstLang, searchTerm string, limit int) ([]*model.Translation, *model.AppError) {
+	ftsQuery := s.getQueryBuilder().
+		Select("object_type", "object_id", "dst_lang", "text", "confidence", "meta").
+		Column("ts_rank(to_tsvector('simple', COALESCE(content_search_text, text)), plainto_tsquery('simple', ?)) AS score", searchTerm).
+		From("translations").
+		Where(sq.Eq{"dst_lang": dstLang}).
+		Where("to_tsvector('simple', COALESCE(content_search_text, text)) @@ plainto_tsquery('simple', ?)", searchTerm).
+		OrderBy("score DESC").
+		Limit(uint64(limit))
+
+	queryString, args, err := ftsQuery.ToSql()
+	if err != nil {
+		return nil, model.NewAppError("SqlAutoTranslationStore.Search",
+			"store.sql_autotranslation.query_build_error", nil, err.Error(), 500)
+	}
+
+	type searchResult struct {
+		ObjectType string          `db:"object_type"`
+		ObjectID   string          `db:"object_id"`
+		DstLang    string          `db:"dst_lang"`
+		Text       string          `db:"text"`
+		Confidence *float64        `db:"confidence"`
+		Score      float64         `db:"score"`
+		Meta       TranslationMeta `db:"meta"`
+	}
+
+	var results []searchResult
+	if err := s.GetReplica().Select(&results, queryString, args...); err != nil {
+		return nil, model.NewAppError("SqlAutoTranslationStore.Search",
+			"store.sql_autotranslation.search.app_error", nil, err.Error(), 500)
+	}
+
+	// Fallback to trigram similarity search if FTS returns no results
+	if len(results) == 0 {
+		trigramQuery := s.getQueryBuilder().
+			Select("object_type", "object_id", "dst_lang", "text", "confidence").
+			Column("similarity(COALESCE(content_search_text, text), ?) AS score", searchTerm).
+			From("translations").
+			Where(sq.Eq{"dst_lang": dstLang}).
+			Where("COALESCE(content_search_text, text) ILIKE '%' || ? || '%'", searchTerm).
+			OrderBy("score DESC").
+			Limit(uint64(limit))
```

Comment:
For my own understanding: in what scenarios would the second query return results while the first didn't?